cmd/gateway/azure/gateway-azure.go

```diff
@@ -834,29 +861,36 @@ func (a *azureObjects) CopyObject(ctx context.Context, srcBucket, srcObject, des
 	if srcOpts.CheckCopyPrecondFn != nil && srcOpts.CheckCopyPrecondFn(srcInfo, "") {
 		return minio.ObjectInfo{}, minio.PreConditionFailed{}
 	}
-	srcBlobURL := a.client.GetContainerReference(srcBucket).GetBlobReference(srcObject).GetURL()
-	destBlob := a.client.GetContainerReference(destBucket).GetBlobReference(destObject)
+	srcBlobURL := a.client.NewContainerURL(srcBucket).NewBlobURL(srcObject).URL()
+	destBlob := a.client.NewContainerURL(destBucket).NewBlobURL(destObject)
 	azureMeta, props, err := s3MetaToAzureProperties(ctx, srcInfo.UserDefined)
 	if err != nil {
 		return objInfo, azureToObjectError(err, srcBucket, srcObject)
 	}
-	destBlob.Metadata = azureMeta
-	err = destBlob.Copy(srcBlobURL, nil)
+	res, err := destBlob.StartCopyFromURL(ctx, srcBlobURL, azureMeta, azblob.ModifiedAccessConditions{}, azblob.BlobAccessConditions{})
 	if err != nil {
 		return objInfo, azureToObjectError(err, srcBucket, srcObject)
 	}
+	copyStatus := res.CopyStatus()
+	for copyStatus != azblob.CopyStatusSuccess {
```

Comment:
Could you add comments why we check for copy status in a loop here ?

(is destBlob.StartCopyFromURL an asynchronous operation ?)

Reply:
Yes, the copy-blob operation is asynchronous so we have to poll for completion ([see docs](https://docs.microsoft.com/en-us/rest/api/storageservices/copy-blob#remarks)). I added a comment to this regard in 0a9aefb.